lib/rex/post/meterpreter/client_core.rb

```diff
@@ -222,6 +248,80 @@ def use(mod, opts = { })
     return true
   end
 
+  def machine_id
```

Comment:
We should probably handle cases where no id is available better (Java/Android/PHP) by coming up with something ID-like (hostname alone?) if the core_machine_id command returns an error. TODO for the future though.